src/snowflake/connector/cursor.py

```diff
@@ -117,6 +117,19 @@ class ResultMetadata(NamedTuple):
     scale: int
     is_nullable: bool
 
+    @classmethod
+    def from_column(cls, col):
+        type_value = FIELD_NAME_TO_ID[col["type"].upper()]
+        return cls(
+            col["name"],
+            type_value,
+            None,
+            col["length"],
+            col["precision"],
+            col["scale"],
+            col["nullable"],
+        )
+
```

Comment:
Using a class method so that its easier to see how the arguments line up with the class fields

Reply:
Maybe for the future if we find ourselves leaving a comment like this that might mean that adding a docstring saying the same thing could be beneficial